metrics/performance/first-meaningful-paint.js

```
@@ -33,41 +33,34 @@ class FirstMeaningfulPaint {
-      // CommitLoad == the first bytes of HTML are returned and Chrome considers
-      //   the navigation a success. A 'isMainFrame' boolean is attached to those events
-      //   However, that flag may be incorrect now, so we're ignoring it.
-      const navStart = userTiming.filter(e => {
-        return e.name === 'navigationStart' && e.args.frame === frameID;
-      })[0];
-
-      // firstContentfulPaint == the first time that text or image content was
-      // painted. See src/third_party/WebKit/Source/core/paint/PaintTiming.h
-      const conPaint = userTiming.filter(e => {
-        return e.name === 'firstContentfulPaint' && e.args.frame === frameID;
-      })[0];
+        .forEach(event => {
+          // navigationStart == the network begins fetching the page URL
+          // CommitLoad == the first bytes of HTML are returned and Chrome considers
+          //   the navigation a success. A 'isMainFrame' boolean is attached to those events
+          //   However, that flag may be incorrect now, so we're ignoring it.
+          if (event.name === 'navigationStart' && !navigationStart) {
+            mainFrameID = event.args.frame;
+            navigationStart = event;
+          }
+          // firstContentfulPaint == the first time that text or image content was
+          // painted. See src/third_party/WebKit/Source/core/paint/PaintTiming.h
+          if (event.name === 'firstContentfulPaint' && event.args.frame === mainFrameID) {
+            firstContentfulPaint = event;
+          }
+        });
 
       // report the raw numbers
-      if (conPaint && navStart) {
-        const navigationStart = navStart.startTime;
-        const firstMeaningfulPaint = conPaint.startTime;
-
+      if (firstContentfulPaint && navigationStart) {
         return resolve({
-          navigationStart,
-          firstMeaningfulPaint
+          navigationStart: navigationStart.startTime,
+          firstMeaningfulPaint: firstContentfulPaint.startTime
```

this is some Closure bug, unless I'm just missing something. Manually casting will silence the errors:
```
navigationStart: /** @type {number} */ (navigationStart.startTime),
firstMeaningfulPaint: /** @type {number} */ (firstContentfulPaint.startTime)
```